### PR TITLE
server/storage/backend: restore original bolt db options after defrag

### PR DESCRIPTION
**Pull request by lavacat**
Problem: Defrag was implemented before custom bolt options were added.
Currently defrag doesn't restore backend options.
For example BackendFreelistType will be unset after defrag.

Solution: save bolt db options and use them in defrag.

fixes https://github.com/etcd-io/etcd/issues/13678

### CODE REVIEW

**Comment by serathius** [comment]
cc @ptabor

File: server/storage/backend/backend_test.go [hunk 1]
```diff
@@ -122,7 +122,16 @@ func TestBackendBatchIntervalCommit(t *testing.T) {
 }
 
 func TestBackendDefrag(t *testing.T) {
-	b, _ := betesting.NewDefaultTmpBackend(t)
+	bcfg := backend.DefaultBackendConfig()
+	// The goal is to verify that we restore config option after defrag.
+	// Just in case check that we aren't setting default value.
+	if bcfg.BackendFreelistType == bolt.FreelistMapType {
+		t.Error("don't use default BackendFreelistType value")
```
Comment (ahrtr) [inline server/storage/backend/backend_test.go:129]: It's not good to make any assumption on the default value. You can just set a different value in the test and verify it after the defragmentation operation.

**Comment by ahrtr** [comment]
It's a really good catch, thanks @lavacat for the fix. Please fix the comment for the unit test.  Overall it looks good to me.

**Review by ahrtr — APPROVED** [review]
LGTM. cc @serathius @ptabor @spzala

**Review by serathius — APPROVED** [review]
LGTM

**Review by spzala — APPROVED** [review]
Thanks @lavacat , makes sense. Running workflows.

**Pull request merged**